src/website/webview/views.py

```diff
@@ -597,7 +610,18 @@ def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
             elif new_status == "pending" and undo_status_change:
                 suggestion.status = CVEDerivationClusterProposal.Status.PENDING
 
-            if old_status != new_status:
+            suggestion.save()
+
+        if status_change and new_status == "published":
+            with transaction.atomic():
+                tracker_issue = suggestion.create_nixpkgs_issue()
+                tracker_issue_link = request.build_absolute_uri(
+                    reverse("webview:issue_detail", args=[tracker_issue.code])
+                )
+                gh_issue_link = create_gh_issue(
+                    cached_suggestion, tracker_issue_link
+                ).html_url
+                suggestion.status = CVEDerivationClusterProposal.Status.PUBLISHED
```

Comment:
The catch here is that the side-effect of creating a git hub issue can't obviously be rolled back, so it could happen that we fail to update some internal stuff and the transaction fails but the GH issue is still created. Though in practice, there is a much higher chance that the failing step is the GH issue creation rather than some internal DB updates.